File: exercism/c/isogram/src/isogram.c
#include <stdbool.h>
#include "isogram.h"

bool isIsogram(char* str) {
    int bs = 0;  // bit set
    char c;
    while ((c = *str++) != '\0') {
        if (c >= 'A' && c <= 'Z')
            c -= 'A';
        else if (c >= 'a' && c <= 'z')
            c -= 'a';
        else continue;
        if ((bs & (1 << c)) > 0) return false;
        bs |= (1 << c);
    }
    return true;
}

File: exercism/go/isogram/isogram.go
package isogram

import "unicode"

// IsIsogram determines whether the given string is an isogram
func IsIsogram(letters string) bool {
	var bset, b uint32
	for _, ch := range letters {
		ch = unicode.ToLower(ch)
		if !unicode.IsLower(ch) {
			continue
		}
		b = 1 << (uint)(ch-'a')
		if (bset & b) > 0 {
			return false
		}
		bset |= b
	}
	return true
}
